pkgs/development/libraries/geogram/default.nix

```diff
@@ -0,0 +1,110 @@
+{ lib
+, stdenv
+, fetchurl
+, fetchFromGitHub
+
+, cmake
+, doxygen
+, zlib
+, python3Packages
+}:
+
+let
+  testdata = fetchFromGitHub {
+    owner = "BrunoLevy";
+    repo = "geogram.data";
+    rev = "70c51a4968960e2d4fa652421b5721672f2b2b70";
+    hash = "sha256-vJtbDDjt2t/6TH0k3i287yGZRd1+qbw20F8D8CFgb3c=";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "geogram";
+  version = "1.8.3";
+
+  src = fetchurl {
```

Comment:
is there a reason you did not use `fetchFromGitHub` here?

Reply:
Geogram requires submodules, but they provide a custom tarball (not one of the autogenerated Github ones) containing the sources including all submodules. This is a bit faster than `fetchFromGitHub` with submodules enabled. I have no strong preferences either way.